src/Symfony/Component/EventDispatcher/InvokableEventSubscriberTrait.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\EventDispatcher;
+
+/**
+ * Helper trait for single-event subscription.
+ *
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ */
+trait InvokableEventSubscriberTrait
+{
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedEvents(): array
+    {
+        return [self::getEventName() => ['__invoke', self::getPriority()]];
+    }
+
+    private static function getEventName(): string
+    {
+        $method = new \ReflectionMethod(self::class, '__invoke');
+
+        if (0 === $method->getNumberOfParameters()) {
+            throw new \LogicException(sprintf('The Event name cannot be inferred from "%s::%s()" method without parameters, you must to define one or implement the "%s::getEventName()" method to return it explicitly.', self::class, $method->getName(), self::class));
```

Comment:
typo: `you must to define`

Reply:
Do we really have use cases where you don't need the event? Looks like rather rare to me. Wouldn't it be better to recommend adding the Event as an argument?

Reply:
Reworked, see latest changes.